To: vim-dev@vim.org
Subject: Patch 7.3.026
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.026
Problem:    CTRL-] in a help file doesn't always work. (Tony Mechelynck)
Solution:   Don't escape special characters. (Carlo Teubner)
Files:	    src/normal.c


*** ../vim-7.3.025/src/normal.c	2010-08-15 21:57:28.000000000 +0200
--- src/normal.c	2010-10-13 18:01:03.000000000 +0200
***************
*** 5666,5673 ****
  	else if (cmdchar == '#')
  	    aux_ptr = (char_u *)(p_magic ? "/?.*~[^$\\" : "/?^$\\");
  	else if (tag_cmd)
! 	    /* Don't escape spaces and Tabs in a tag with a backslash */
! 	    aux_ptr = (char_u *)"\\|\"\n[";
  	else
  	    aux_ptr = (char_u *)"\\|\"\n*?[";
  
--- 5666,5678 ----
  	else if (cmdchar == '#')
  	    aux_ptr = (char_u *)(p_magic ? "/?.*~[^$\\" : "/?^$\\");
  	else if (tag_cmd)
! 	{
! 	    if (curbuf->b_help)
! 		/* ":help" handles unescaped argument */
! 		aux_ptr = (char_u *)"";
! 	    else
! 		aux_ptr = (char_u *)"\\|\"\n[";
! 	}
  	else
  	    aux_ptr = (char_u *)"\\|\"\n*?[";
  
*** ../vim-7.3.025/src/version.c	2010-10-13 17:50:02.000000000 +0200
--- src/version.c	2010-10-13 18:04:54.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     26,
  /**/

-- 
MAN:    Fetchez la vache!
GUARD:  Quoi?
MAN:    Fetchez la vache!
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
